hkm/views/views.py

```diff
@@ -374,19 +374,28 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class HomeView(BaseCollectionListView):
+class HomeView(BaseView):
     template_name = 'hkm/views/home_page.html'
     url_name = 'hkm_index'
 
+    collection_qs = Collection.objects.none()
+    showcase_qs = Showcase.objects.none()
+
+    def setup(self, request, *args, **kwargs):
+        self.collection_qs = self.get_collection_qs(request, *args, **kwargs)
+        self.showcase_qs = self.get_showcase_qs(request, *args, **kwargs)
+        return True
+
     def get_collection_qs(self, request, *args, **kwargs):
-        return Collection.objects.filter(is_public=True).order_by('created')
+        return Collection.objects.filter(show_in_landing_page=True)
+
+    def get_showcase_qs(self, request, *args, **kwargs):
+        return Showcase.objects.filter(show_on_home_page=True)
 
     def get_context_data(self, **kwargs):
         context = super(HomeView, self).get_context_data(**kwargs)
-        context['frontpage_images'] = self.collection_qs.filter(
-        show_in_landing_page=True)
-        context['featured_collections'] = self.collection_qs.filter(
-            is_featured=True)
+        context['frontpage_images'] = self.collection_qs.order_by('created')
+        context['showcases'] = self.showcase_qs
         return context
```

Comment:
About the whole view, as it doesn't inherit from `BaseCollectionListView` anymore, IMO there is no point to handle the querysets so complexly, I would do just something like
```
class HomeView(BaseView):
    template_name = 'hkm/views/home_page.html'
    url_name = 'hkm_index'

    def get_context_data(self, **kwargs):
        context = super(HomeView, self).get_context_data(**kwargs)
        context['frontpage_images'] = Collection.objects.filter(show_in_landing_page=True).order_by('created')
        context['showcases'] = Showcase.objects.filter(show_on_home_page=True)
```

Also FWIW the showcases don't have ordering applied, so they will be in random order.

Reply:
That's so much simpler, seems Im little more rusty than I though. I moved `order_by` to showcases, thats where it was supposed to be in the first place.

Reply:
Ok good, but now the collections / frontpage images seem to be in random order? :)

Reply:
Thats ok, i'm taking one of the pictures from that list at random at showing it on the frontpage :)